lib/output.js

```diff
@@ -1286,8 +1286,16 @@ function OutputStream(options) {
 
     // self should be AST_New.  decide if we want to show parens or not.
     function need_constructor_parens(self, output) {
-        // Always print parentheses with arguments
-        return self.args.length > 0;
+         // Always print parentheses with arguments
+         if (self.args.length > 0) return true;
+
+        // Certain expressions are safe to add constructor calling parentheses
+        return output.option("beautify") && (
```

Comment:
Hmm I can actually follow you logic, the real fix in #1111 was actually the parentheses around `!0`